### ISSUE TITLE
Couldn’t communicate with a helper application.

### ISSUE THREAD

**Issue by openemu1** [OpenEmu/OpenEmu#2114]
A bunch of arcade roms were working great and then one day they stopped opening.  When i click them it says:  "Couldn’t communicate with a helper application. Try your operation again. If that fails, quit and relaunch the application and try again."

I tried uninstalling openemu and re-installing and getting new roms and still it does not work.  HELP!!!

**Comment by openemu1** [comment]
the sample demo games work.  the Arcade games do not!!

**Comment by mrvacbob** [comment]
"Couldn’t communicate with a helper application" means there was a crash in the background. Please look for and attach any related files from ~/Library/Logs/DiagnosticReports/.

**Comment by openemu1** [comment]
I was trying to play Afterburner game in Arcade.  Let me know if this info helps.

SparkelUpdatelog.log says:

2015-10-26 05:14:28 +0000: ===== OpenEmu 3 =====

system.log says:


Oct 26 01:14:25 GeraldGeinsiMac OpenEmu[441]: OELibraryDatabase loadFromURL: 'file:///Users/generg/Library/Application%20Support/OpenEmu/Game%20Library/'
Oct 26 01:14:25 GeraldGeinsiMac OpenEmu[441]: CoreData: Failed to load optimized model at path '/Applications/OpenEmu 3.app/Contents/Resources/OEDatabase.momd/OEDatabase 1.3.omo'
Oct 26 01:14:26 GeraldGeinsiMac OpenEmu[441]: Check for updates (108786.210903 > 86400.000000)
Oct 26 01:14:28 GeraldGeinsiMac OpenEmu[441]: Apple Keyboard
Oct 26 01:14:30 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:14:31 --- last message repeated 1 time ---
Oct 26 01:14:31 GeraldGeinsiMac OpenEmu[441]: Cannot find executable for CFBundle 0x7fd585078ef0 </Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin> (not loaded)
Oct 26 01:14:31 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:14:31 --- last message repeated 2 times ---
Oct 26 01:14:31 GeraldGeinsiMac appleeventsd[49]: SecTaskLoadEntitlements failed error=22
Oct 26 01:14:31 GeraldGeinsiMac OpenEmuHelperApp[445]: parent application is: OpenEmu
Oct 26 01:14:31 GeraldGeinsiMac OpenEmuHelperApp[445]: Cannot find executable for CFBundle 0x7ff4b164b8b0 </Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin> (not loaded)
Oct 26 01:14:31 GeraldGeinsiMac OpenEmuHelperApp[445]: ROM did not load.
Oct 26 01:14:31 GeraldGeinsiMac OpenEmuHelperApp[445]: <NSXPCConnection: 0x7ff4b1532b30> connection from pid 441: Warning: Exception caught during invocation of received message, dropping incoming message and invalidating the connection.
    Exception: *** -[NSProxy methodSignatureForSelector:] called!
Oct 26 01:14:31 GeraldGeinsiMac OpenEmu[441]: Helper Connection (0x7fd5857a9690) failed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection from pid 0" UserInfo={NSDebugDescription=connection from pid 0}
Oct 26 01:14:33 GeraldGeinsiMac OpenEmu[441]: Deallocating proxy: <OEThreadProxy 0x7fd5878374c0, target: <OEGameDocument 0x7fd58577c620, ROM: 'sf2cej', System: 'openemu.system.arcade', Core: 'org.openemu.MAME'>>
Oct 26 01:14:34 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:14:34 --- last message repeated 2 times ---
Oct 26 01:14:34 GeraldGeinsiMac appleeventsd[49]: SecTaskLoadEntitlements failed error=22
Oct 26 01:14:34 GeraldGeinsiMac OpenEmuHelperApp[447]: parent application is: OpenEmu
Oct 26 01:14:34 GeraldGeinsiMac OpenEmuHelperApp[447]: Cannot find executable for CFBundle 0x7faca94397d0 </Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin> (not loaded)
Oct 26 01:14:34 GeraldGeinsiMac OpenEmuHelperApp[447]: ROM did not load.
Oct 26 01:14:34 GeraldGeinsiMac OpenEmuHelperApp[447]: <NSXPCConnection: 0x7faca952b840> connection from pid 441: Warning: Exception caught during invocation of received message, dropping incoming message and invalidating the connection.
    Exception: *** -[NSProxy methodSignatureForSelector:] called!
Oct 26 01:14:34 GeraldGeinsiMac OpenEmu[441]: Helper Connection (0x7fd587840340) failed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection from pid 0" UserInfo={NSDebugDescription=connection from pid 0}
Oct 26 01:14:36 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:14:36 GeraldGeinsiMac OpenEmu[441]: Deallocating proxy: <OEThreadProxy 0x7fd582f7e050, target: <OEGameDocument 0x7fd5857c8e50, ROM: 'After Burner II', System: 'openemu.system.arcade', Core: 'org.openemu.MAME'>>


 

Oct 26 01:28:16 GeraldGeinsiMac OpenEmu[441]: Deallocating proxy: <OEThreadProxy 0x7fd58564ee40, target: <OEGameDocument 0x7fd585798950, ROM: 'After Burner II', System: 'openemu.system.arcade', Core: 'org.openemu.MAME'>>
Oct 26 01:28:16 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:20 --- last message repeated 3 times ---
Oct 26 01:28:20 GeraldGeinsiMac appleeventsd[49]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:20 GeraldGeinsiMac OpenEmuHelperApp[475]: parent application is: OpenEmu
Oct 26 01:28:20 GeraldGeinsiMac OpenEmuHelperApp[475]: Some shit about the game.
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: Ines: 128k PRG-ROM set
    Ines: vertical mirroring set
    Ines: NTSC set
    Ines: mapper 2 set
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: Board: UNROM
    Board: 128k PRG-ROM
    Board: 8k V-RAM
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: Load returned : 0
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: Power on returned : 0
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: Setting up emulation
Oct 26 01:28:21 GeraldGeinsiMac OpenEmu[441]: SETUP DONE.
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: Begin separate thread
Oct 26 01:28:21 GeraldGeinsiMac OpenEmuHelperApp[475]: RT policy: 0.007000s (limit 0.030000s) every 0.016667s
Oct 26 01:28:21 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:22 GeraldGeinsiMac coreaudiod[130]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:22 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:22 GeraldGeinsiMac OpenEmuHelperApp[475]: Synchronizing because we are 1.06033 seconds behind
Oct 26 01:28:30 GeraldGeinsiMac OpenEmuHelperApp[475]: Power off returned : 0
Oct 26 01:28:30 GeraldGeinsiMac OpenEmuHelperApp[475]: Unload returned : 0
Oct 26 01:28:30 GeraldGeinsiMac OpenEmuHelperApp[475]: Did finish separate thread
Oct 26 01:28:30 GeraldGeinsiMac OpenEmu[441]: Deallocating proxy: <OEThreadProxy 0x7fd587808d70, target: <OEGameDocument 0x7fd58782e440, ROM: 'streemerz', System: 'openemu.system.nes', Core: 'org.openemu.Nestopia'>>

Oct 26 01:28:45 GeraldGeinsiMac OpenEmuHelperApp[478]: parent application is: OpenEmu
Oct 26 01:28:45 GeraldGeinsiMac OpenEmuHelperApp[478]: Cannot find executable for CFBundle 0x7fcd33f3cad0 </Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin> (not loaded)
Oct 26 01:28:45 GeraldGeinsiMac OpenEmuHelperApp[478]: ROM did not load.
Oct 26 01:28:45 GeraldGeinsiMac OpenEmuHelperApp[478]: <NSXPCConnection: 0x7fcd33c12ac0> connection from pid 441: Warning: Exception caught during invocation of received message, dropping incoming message and invalidating the connection.
    Exception: *** -[NSProxy methodSignatureForSelector:] called!
Oct 26 01:28:45 GeraldGeinsiMac OpenEmu[441]: Helper Connection (0x7fd5851a9660) failed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection from pid 0" UserInfo={NSDebugDescription=connection from pid 0}
Oct 26 01:28:46 GeraldGeinsiMac OpenEmu[441]: Deallocating proxy: <OEThreadProxy 0x7fd587829240, target: <OEGameDocument 0x7fd582c436b0, ROM: 'sf2cej', System: 'openemu.system.arcade', Core: 'org.openemu.MAME'>>
Oct 26 01:28:47 GeraldGeinsiMac launchservicesd[74]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:47 --- last message repeated 2 times ---
Oct 26 01:28:47 GeraldGeinsiMac appleeventsd[49]: SecTaskLoadEntitlements failed error=22
Oct 26 01:28:47 GeraldGeinsiMac OpenEmuHelperApp[480]: parent application is: OpenEmu
Oct 26 01:28:47 GeraldGeinsiMac OpenEmuHelperApp[480]: Cannot find executable for CFBundle 0x7f8fda42f8e0 </Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin> (not loaded)
Oct 26 01:28:47 GeraldGeinsiMac OpenEmuHelperApp[480]: ROM did not load.
Oct 26 01:28:47 GeraldGeinsiMac OpenEmuHelperApp[480]: <NSXPCConnection: 0x7f8fda616d90> connection from pid 441: Warning: Exception caught during invocation of received message, dropping incoming message and invalidating the connection.
    Exception: *** -[NSProxy methodSignatureForSelector:] called!
Oct 26 01:28:47 GeraldGeinsiMac OpenEmu[441]: Helper Connection (0x7fd587939fe0) failed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection from pid 0" UserInfo={NSDebugDescription=connection from pid 0}
Oct 26 01:28:49 GeraldGeinsiMac OpenEmu[441]: Deallocating proxy: <OEThreadProxy 0x7fd5851a8b10, target: <OEGameDocument 0x7fd58783bab0, ROM: 'After Burner II', System: 'openemu.system.arcade', Core: 'org.openemu.MAME'>>
 

Date: Sun, 25 Oct 2015 01:43:38 -0700

**Comment by mrvacbob** [comment]
```
Oct 26 01:14:31 GeraldGeinsiMac OpenEmuHelperApp[445]: <NSXPCConnection: 0x7ff4b1532b30> connection from pid 441: Warning: Exception caught during invocation of received message, dropping incoming message and invalidating the connection.
    Exception: *** -[NSProxy methodSignatureForSelector:] called!
Oct 26 01:28:47 GeraldGeinsiMac OpenEmuHelperApp[480]: Cannot find executable for CFBundle 0x7f8fda42f8e0 </Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin> (not loaded)

```

**Comment by openemu1** [comment]
what do you suggest i do?

Date: Sun, 25 Oct 2015 23:41:23 -0700

**Comment by openemu1** [comment]
Your last post showed some code but did not say what I should do.  How can I fix so Arcade will work?

**Comment by mrvacbob** [comment]
You probably can't.

**Comment by openemu1** [comment]
What is causing the problem?

**Comment by openemu1** [comment]
Why is this happening?  Why did Arcade work and then all of a sudden stop working?  What is causing the problem?

**Comment by clobber** [comment]
>Why is this happening?

No idea what you did.

>Cannot find executable for CFBundle 0x7f8fda42f8e0 /Users/generg/Library/Application Support/OpenEmu/Cores/MAME.oecoreplugin (not loaded)

Means the the plugin is not there.

1. Make sure you have the latest OpenEmu (in your case, version 1.0.4 'experimental' cores bundle)
2. Navigate to the folder `~/Library/Application Support/OpenEmu` (~ meaning your Home folder where ~ is your username. Guide here if you don't know this: http://osxdaily.com/2014/12/16/show-user-library-folder-os-x-yosemite/)
3. Delete the "Systems" folder (if it exists)
4. Delete the "Cores" folder
5. Restart your computer
6. Reopen OpenEmu
7. Navigate to OpenEmu > Preferences > Cores, click download for all the cores so they redownload

**Comment by openemu1** [comment]
IT WORKED!!!!  THANK YOU SO MUCH CLOBBER!!!!!

**Comment by thewafflefactory** [comment]
Good lord I am at the mercy of those on the internet! Today I dragged a rom into OpenEmu and that made it crash every time i click any game, no matter what game. I uninstalled and redownloaded the cores, no go. I followed the directions listed above, no go. I did a deep uninstall with an app from the app store just to make sure it was a clean ass install and now it does run some random Game boy games but a majority now says "Couldn't Communicate with a helper application." I am at a loss :(
